040_d3_L54/040_d3_L54_safe.py

```
@@ -0,0 +1,154 @@
+        safe_response_data = {k: re.sub(r'[\r\n\t]', '_', str(v)) if isinstance(v, str) else v for k, v in response_data.items()}
+        logging.info(f"Received response from AI: {safe_response_data}")
+
+        ai_message = response_data.get('ai_message', '')
+        ai_score = float(response_data.get('ai_score', 0))
+        new_session_id = response_data.get('SessionId', '')
+        
+        return ai_message, ai_score, new_session_id
+    except requests.HTTPError as http_err:
+        # Sanitize error messages for logging to prevent log injection
+        safe_http_err = re.sub(r'[\r\n\t]', '_', str(http_err))
+        logging.error(f'HTTP error occurred: {safe_http_err}')
```

Ruff (0.14.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)